app/screens/permalink/permalink.js

```diff
@@ -0,0 +1,458 @@
+                navBarTextColor: theme.sidebarHeaderTextColor,
+                navBarBackgroundColor: theme.sidebarHeaderBg,
+                navBarButtonColor: theme.sidebarHeaderTextColor,
+                screenBackgroundColor: theme.centerChannelBg
+            },
+            passProps: {
+                channelId,
+                rootId
+            }
+        };
+
+        navigator.push(options);
+    });
+
+    handleClose = () => {
+        const {actions, navigator, onClose} = this.props;
+        if (this.refs.view) {
+            this.refs.view.zoomOut().then(() => {
+                actions.selectPost('');
+                navigator.dismissModal({animationType: 'none'});
+
+                if (onClose) {
+                    onClose();
+                }
+            });
+        }
+    };
+
+    jumpToChannel = (channelId, channelDisplayName) => {
+        if (channelId) {
+            const {actions, channelTeamId, currentTeamId, navigator, onClose, theme} = this.props;
+            const currentChannelId = this.props.channelId;
+            const {
+                handleSelectChannel,
+                handleTeamChange,
+                markChannelAsRead,
+                setChannelLoading,
+                setChannelDisplayName,
+                markChannelAsViewed
+            } = actions;
+
+            if (this.refs.view) {
+                this.refs.view.zoomOut().then(() => {
+                    actions.selectPost('');
+
+                    if (onClose) {
+                        onClose();
+                    }
+
+                    navigator.resetTo({
+                        screen: 'Channel',
+                        animated: true,
+                        animationType: 'fade',
+                        navigatorStyle: {
+                            navBarHidden: true,
+                            statusBarHidden: false,
+                            statusBarHideWithNavBar: false,
+                            screenBackgroundColor: theme.centerChannelBg
+                        }
+                    });
+
+                    if (channelTeamId && currentTeamId !== channelTeamId) {
+                        handleTeamChange(channelTeamId, false);
+                    }
+
+                    setChannelLoading(channelId !== currentChannelId);
+                    setChannelDisplayName(channelDisplayName);
+                    handleSelectChannel(channelId);
+
+                    InteractionManager.runAfterInteractions(async () => {
+                        markChannelAsRead(channelId, currentChannelId);
+                        if (channelId !== currentChannelId) {
+                            markChannelAsViewed(currentChannelId);
+                        }
+                    });
+                });
+            }
+        }
+    };
+
+    handlePress = () => {
+        const {channelId, channelName} = this.props;
+
+        if (this.refs.view) {
+            this.refs.view.growOut().then(() => {
```

Comment:
It looks like we do these animations in two steps since we `growOut` here and then `zoomOut` above. Are both of those necessary, or should they be done at the same time?